profiling/cloud_profiler.go

```diff
@@ -16,32 +18,70 @@ type GoogleProfilerFlag struct{}
+		log.Printf("%+v", errors.WithStack(fmt.Errorf("pkg/profiling: GoogleProfilerFlag.Set: missing/blank required env var EMPIRE_PROCESS")))
+
+		return nil
+	}
+
+	empire_release := os.Getenv("EMPIRE_RELEASE")
+	if empire_release == "" {
+		log.Printf("%+v", errors.WithStack(fmt.Errorf("pkg/profiling: GoogleProfilerFlag.Set: missing/blank required env var EMPIRE_RELEASE")))
+
+		return nil
+	}
+
 	creds := os.Getenv("GOOGLE_APPLICATION_CREDENTIALS_CONTENT")
+	if creds == "" {
+		log.Printf("%+v", errors.WithStack(fmt.Errorf("pkg/profiling: GoogleProfilerFlag.Set: missing/blank required env var GOOGLE_APPLICATION_CREDENTIALS_CONTENT")))
+
+		return nil
+	}
+
 	credsPath := os.Getenv("GOOGLE_APPLICATION_CREDENTIALS")
+	if credsPath == "" {
+		log.Printf("%+v", errors.WithStack(fmt.Errorf("pkg/profiling: GoogleProfilerFlag.Set: missing/blank required env var GOOGLE_APPLICATION_CREDENTIALS")))
 
-	if creds != "" {
-		// kludge alert! Our 12-factor way of deploying applications can't pass
-		// configuration to programs except through environment variables, but
-		// the GCP SDK expects the credentials to be in a file at the path of
-		// $GOOGLE_APPLICATION_CREDENTIALS. So, we put the actual credentials
-		// in a different environment variable, and write them to a file.
-		//
-		// If $GOOGLE_APPLICATION_CREDENTIALS_CONTENT isn't set then we do
-		// nothing, since we might be running in an environment where the
-		// credentials might be discovered through other means.
+		return nil
+	}
+
+	// kludge alert! Our 12-factor way of deploying applications can't pass
+	// configuration to programs except through environment variables, but
+	// the GCP SDK expects the credentials to be in a file at the path of
+	// $GOOGLE_APPLICATION_CREDENTIALS. So, we put the actual credentials
+	// in a different environment variable, and write them to a file.
+	//
+	// If $GOOGLE_APPLICATION_CREDENTIALS_CONTENT isn't set then we do
+	// nothing, since we might be running in an environment where the
+	// credentials might be discovered through other means.
+	if err := ioutil.WriteFile(credsPath, []byte(creds), 0600); err != nil {
+		log.Printf("%+v", errors.WithStack(fmt.Errorf("pkg/profiling: GoogleProfilerFlag.Set: could not write $GOOGLE_APPLICATION_CREDENTIALS_CONTENT to %s: %s", credsPath, err)))
 
-		if err := ioutil.WriteFile(credsPath, []byte(creds), 0600); err != nil {
-			return errors.Wrap(err, "could not write $GOOGLE_APPLICATION_CREDENTIALS_CONTENT to "+credsPath)
-		}
+		return nil
 	}
 
 	cfg := profiler.Config{
-		Service:        os.Getenv("EMPIRE_APPNAME") + "." + os.Getenv("EMPIRE_PROCESS"),
-		ServiceVersion: os.Getenv("EMPIRE_RELEASE"),
+		Service:        fmt.Sprintf("%s.%s", empire_appname, empire_process),
+		ServiceVersion: empire_release,
 		ProjectID:      projectID,
 	}
 
-	return profiler.Start(cfg)
+	err := profiler.Start(cfg)
+	if err != nil {
+		log.Printf("pkg/profiling: GoogleProfilerFlag.Set: error starting profiler: %s", err)
```

Comment:
Might as well do `WithStack` and `%+v` here as well

Reply:
Whoops, yeah, meant to.